### PR TITLE
feat: #199 Add button on WayPage

### PR DESCRIPTION
**Pull request by dayanych**
WayPage
- [x] add button with value "Add to favorite" (on click the will be added to favorite on UserPage)
- [x] button not appears for guests (not logged in users)
- [x] if user already has this way in favorite - this button has value "Remove from favorite" (on  click it will be removed from favorite ways)
- [ ] add to the Way model the field "favoriteUserUuids"
- [ ] add column "favorite" for each table with waysPreview and on Way Page (somewhere above, we will structure it later)
- [ ] after click on button "Add to favorite" in all tables where column  "favorite" exist number should increased
- [ ]  after click on button "Remove from favorite" in all tables where column  "favorite" exist number should decrease

### CODE REVIEW

File: mw-webapp/_tmp_9684_a467567fe74ca72b31e9bd6afe2ca25e
Comment (tritonJS826) [inline mw-webapp/_tmp_9684_a467567fe74ca72b31e9bd6afe2ca25e:1]: whg do we need this file?

File: mw-webapp/src/dataAccessLogic/WayPreviewDAL.ts [hunk 1]
```diff
@@ -55,7 +55,7 @@ export class WayPreviewDAL {
     const newGoal = await GoalPreviewDAL.createGoalPreview();
 
     const DEFAULT_WAY: WayDTOWithoutUuid = {
-      name: `${DateUtils.getShortISODateValue(new Date)} Way of ${user.name}`,
+      name: `${DateUtils.getShortISODateValue(new Date())} Way of ${user.name}`,
```
Comment (tritonJS826) [inline mw-webapp/src/dataAccessLogic/WayPreviewDAL.ts:58]: why

File: mw-webapp/src/logic/wayPage/WayPage.tsx [hunk 2]
```diff
@@ -50,9 +77,20 @@ export const WayPage = (props: WayPageProps) => {
...
 
+  /**
+   * Load current user
+   */
+  const loadCurrentUser = async () => {
+    if (user) {
+      const currentUserPreview = await UserPreviewDAL.getUserPreview(user.uid);
+      setCurrentUser(currentUserPreview);
+    }
+  };
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/WayPage.tsx:88]: Why? check video

File: mw-webapp/src/logic/wayPage/WayPage.tsx [hunk 3]
```diff
@@ -50,9 +77,20 @@ export const WayPage = (props: WayPageProps) => {
...
+
   useEffect(() => {
     loadWay();
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/WayPage.tsx:91]: could we make it pour? check video

File: mw-webapp/src/logic/wayPage/WayPage.tsx [hunk 4]
```diff
@@ -50,9 +77,20 @@ export const WayPage = (props: WayPageProps) => {
...
-  }, []);
+    loadCurrentUser();
+  }, [way]);
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/WayPage.tsx:93]: why? check video

File: mw-webapp/src/logic/wayPage/WayPage.tsx [hunk 5]
```diff
@@ -96,6 +148,10 @@ export const WayPage = (props: WayPageProps) => {
             onChangeFinish={(description) => updateGoalWay(way, description)}
             rows={5}
           />
+          {currentUser && <Button
+            value={isWayInFavorites(way.uuid, currentUser.favoriteWays) ? "Remove from favorite" : "Add to favorite"}
+            onClick={() => updateFavoriteWays(way.uuid, currentUser.favoriteWays)}
+          />}
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/WayPage.tsx:154]: chek video

**Review by tritonJS826 — COMMENTED** [review]
check PR description, looks like wrong checkmarks

File: mw-webapp/src/dataAccessLogic/WayPreviewDAL.ts (hunk 1 above)
Comment (dayanych) [reply mw-webapp/src/dataAccessLogic/WayPreviewDAL.ts]: Deleted.

File: mw-webapp/src/logic/wayPage/WayPage.tsx (hunk 4 above)
Comment (dayanych) [reply mw-webapp/src/logic/wayPage/WayPage.tsx]: Done.

File: mw-webapp/src/logic/wayPage/WayPage.tsx (hunk 3 above)
Comment (dayanych) [reply mw-webapp/src/logic/wayPage/WayPage.tsx:107]: Done.

File: mw-webapp/src/logic/wayPage/WayPage.tsx (hunk 2 above)
Comment (dayanych) [reply mw-webapp/src/logic/wayPage/WayPage.tsx:104]: Done.

**Review by Ekaterina1994 — APPROVED** [review]

**Review by tritonJS826 — APPROVED** [review]

**Pull request merged**